### PR TITLE
chore(ci): verify commit on release

### PR DESCRIPTION
**Pull request by soonum**
Enforce commit being associated to a tag.
The tag must be committed by a member of the release team. In addition, the tag needs to be verified. Finally, triggering actor must also be a member of the release team.

### CODE REVIEW

File: .github/workflows/make_release.yml [hunk 1]
```diff
@@ -30,8 +30,59 @@ env:
...
+        run: |
+          if [ ${{ steps.author_check.outputs.authorized }} = false ]; then
+            echo "Author '${{ env.COMMITTER_LOGIN }}' is not part of authorized team"
+            exit 1
+          fi
+          
+          if [ ${{ steps.actor_check.outputs.authorized }} = false ]; then
+            echo "Actor '${{ env.COMMITTER_LOGIN }}' is not authorized to perform release"
+            exit 1
+          fi
```
Comment (IceTDrinker) [inline .github/workflows/make_release.yml:76]: wrong actor reported here

File: .github/workflows/make_release.yml [hunk 2]
```diff
@@ -30,8 +30,59 @@ env:
...
+          org: ${{ github.repository_owner }}
+          team: ${{ secrets.RELEASE_TEAM }}
+          github_token: ${{ secrets.READ_ORG_TOKEN }}
+
+      - name: Commit verification
+        run: |
+          if [ ${{ steps.author_check.outputs.authorized }} = false ]; then
+            echo "Author '${{ env.COMMITTER_LOGIN }}' is not part of authorized team"
+            exit 1
+          fi
```
Comment (IceTDrinker) [inline .github/workflows/make_release.yml:71]: quote everything here, false is a built in in bash, the output on the left is a string, also the check should be I believe == ?

**Review by IceTDrinker — CHANGES_REQUESTED** [review]
I think the shell scripting has issues, a mistake in a log

**Comment by soonum** [comment]
Should we extend these checks to the rest of release workflows ?
If so I can do it in a another PR but first I'd like to factorize this step into its own workflow and use `workflow_call` event to use it in all release workflows.

**Comment by IceTDrinker** [comment]
> Should we extend these checks to the rest of release workflows ? If so I can do it in a another PR but first I'd like to factorize this step into its own workflow and use `workflow_call` event to use it in all release workflows.

yes we should

**Comment by IceTDrinker** [comment]
could be done via a callable workflow, we have example of those somewhere, I can show you

File: .github/workflows/verify_tagged_commit.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags/')
+    steps:
+      - name: Get commit details
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          {
+            echo "COMMITTER_LOGIN=$(gh api repos/${{ github.repository }}/commits/${{ github.sha }} --jq '.committer.login')";
+            echo "VERIFICATION_STATUS=$(gh api repos/${{ github.repository }}/commits/${{ github.sha }} --jq '.commit.verification.verified')";
```
Comment (IceTDrinker) [inline .github/workflows/verify_tagged_commit.yml:23]: question here does the sha refer to the tag itself ?

Comment (soonum) [reply .github/workflows/verify_tagged_commit.yml:23]: Yes , but in the end I need to modify the URL to get the actual tag.
It needs some fixing, thanks for noticing.

**Comment by soonum** [comment]
Regarding discussion on Slack to the impossibility to get the SHA and the committer login for the tag, we've decided to only check the triggering actor to ensure a certain amount of safety to make a release.

**Review by IceTDrinker — APPROVED** [review]

**Pull request merged**